docs/userguide/deployment.rst

```diff
@@ -49,29 +49,29 @@ A production-ready configuration file should contain at least the following opti
 
 Suppose you had the following component class as your root component::
 
-    class MyRootComponent(ContainerComponent):
+    class MyRootComponent(Component):
         def __init__(self, components, data_directory: str):
             super().__init__(components)
             self.data_directory = data_directory
-
-        async def start() -> None:
             self.add_component('mailer', backend='smtp')
             self.add_component('sqlalchemy')
-            await super().start()
 
 You could then write a configuration file like this::
 
     ---
     max_threads: 20
     component:
       type: !!python/name:myproject.MyRootComponent
-      data_directory: /some/file/somewhere
+      config:
+        data_directory: /some/file/somewhere
       components:
         mailer:
-          host: smtp.mycompany.com
-          ssl: true
+          config:
+            host: smtp.mycompany.com
+            ssl: true
```

Comment:
Are you sure about these changes? It seems it still works the old way:
```
        mailer:
          host: smtp.mycompany.com
          ssl: true
```

Reply:
This was something I was planning but I never implemented it. Good catch.